src/Orleans.Core/Messaging/ClientMessageCenter.cs

```diff
@@ -403,7 +403,14 @@ public void Disconnect()
         /// </summary>
         public void Reconnect()
         {
-            throw new NotImplementedException("Reconnect");
+            //TODO: This is probably not be the proper way to reconnect since the connection to the test cluster is sometimes not regained
+            foreach (var connection in gatewayConnections.Values.ToArray())
+            {
+                if (!connection.IsLive)
+                {
+                    connection.Start();
+                }
```

Comment:
This doesn't seem right to me. @benjaminpetit do you have any thoughts on how this should be implemented or if it should be implemented at all?

Reply:
On the other hand, given it's explicitly for testing only, I think it's fine

Reply:
Currently this wont work because `IsLive` is never set back to `true` among other potential issues (`ThreadState`) inside of `GatewayConnection`.

Reply:
Apologies for the slow reply. I currently don't have access to my Windows machine and it seems not possible to use my Mac to do any proper development on the project, so until I get my Windows machine fixed I can't do any additional work on this pull request. (Sidenote: even on the Windows machine doing a clean checkout would not work properly as a lot of tests kept failing randomly, not sure if there was something wrong with the setup. Would still be nice to have some FAQ on how to setup the project as a new contributor as the steps in the README did not work out of the box and there was a bunch of stuff I had to do to even run a single test)

As for the Reconnect, I had strong suspicions this was not the proper way to do it. It was not easy to figure out what the proper way is (frankly, I still don't know). I tried various approaches to make a new Gateway fire a GatewayCountChanged event with an increased gateway count, but none of them seemed to work. If anyone can point out the proper way to do this then I can fix the code and make the tests pass properly.